src/sgame/components/RocketpodComponent.cpp

```diff
@@ -265,9 +265,13 @@ void RocketpodComponent::Shoot(const glm::vec3& direction) {
 	entity.oldEnt->target = target->oldEnt;
 	G_AddEvent(entity.oldEnt, EV_FIRE_WEAPON, 0);
 
-	gentity_t* missile = G_SpawnMissile(MIS_ROCKET, entity.oldEnt, entity.oldEnt->s.pos.trBase, &direction[0],
-	                                    target->oldEnt, RocketThink, level.time + ROCKET_TURN_PERIOD);
-	missile->timestamp = level.time + BG_Missile(MIS_ROCKET)->lifetime;
+	gentity_t* m = G_NewEntity(HAS_CBSE);
+	RocketMissileEntity::Params params;
+	params.oldEnt = m;
+	params.Missile_attributes = BG_Missile(MIS_ROCKET);
+	m->entity = new RocketMissileEntity{ params };
+	G_SetUpMissile(m, MIS_ROCKET, entity.oldEnt, entity.oldEnt->s.pos.trBase, &direction[0]);
+	m->target = target->oldEnt;
```

Comment:
Should this have a helper function to avoid this boiler plate?

Reply:
Maybe, although it would have to be a template which is annoying.

What if we had the code generator make a builder-pattern struct for the Params thing? Like `RocketMissileEntity::Params().oldEnt(m).Missile_attributes(BG_Missile(MIS_ROCKET))` etc.

Reply:
I think I template is way cleaner and readable.

(probably doesn't compile...)
```cpp
template<classname T>
gentity_t* NewMIssile(mis missle_t) {
        gentity_t* m = G_NewEntity(HAS_CBSE);
	T::Params params;
	params.oldEnt = m;
	params.Missile_attributes = BG_Missile(mis);
	m->entity = new T{ params };
	G_SetUpMissile(m, MIS_ROCKET, entity.oldEnt, entity.oldEnt->s.pos.trBase, &direction[0]);
	m->target = target->oldEnt;
        return m;
}
```

Reply:
It works well as long as all missile types have the same params. But in one of my later commits I have the hive and rocket targets being passed in params, so then you have to pass around half-finished param structs and it's not that nice.